### PR TITLE
Make several hardcoded settings user configurable

### PR DESCRIPTION
**Pull request by jtuple**
This pull-request centralizes all settings / timeout values into the new `riak_ensemble_config` module as well as makes many of these settings user configurable via `app.config`. It's unclear which (if any) of these settings should be cuttlefish'd (nor what the appropriate user-facing names should be).

Addresses #18

### CODE REVIEW

**Comment by andrewjstone** [comment]
- [x] code review
- [x] riak tests

**Comment by andrewjstone** [comment]
:+1: 53cf04b

File: src/riak_ensemble_peer.erl [hunk 1]
```diff
@@ -55,6 +55,17 @@
...
+-define(ALIVE,   riak_ensemble_config:alive_ticks()).
+-define(WORKERS, riak_ensemble_config:peer_workers()).
+
+-define(FOLLOWER_TIMEOUT,  riak_ensemble_config:follower_timeout()).
+-define(PENDING_TIMEOUT,   riak_ensemble_config:pending_timeout()).
+-define(ELECTION_TIMEOUT,  riak_ensemble_config:election_timeout()).
+-define(PREFOLLOW_TIMEOUT, riak_ensemble_config:prefollow_timeout()).
+-define(PROBE_DELAY,       riak_ensemble_config:probe_delay()).
+-define(LOCAL_GET_TIMEOUT, riak_ensemble_config:local_get_timeout()).
+-define(LOCAL_PUT_TIMEOUT, riak_ensemble_config:local_put_timeout()).
```
Comment (reiddraper) [inline src/riak_ensemble_peer.erl:67]: These are all defined at _compile_ time, right? Is that intended?

Comment (jtuple) [inline src/riak_ensemble_peer.erl:67]: The intention is for compile time to replace things like `?FOLLOWER_TIMEOUT` with `riak_ensemble_config:follower_timeout()`, which is call that is then made dynamically at runtime. Some of the functions in `riak_ensemble_config` are hardcoded, others map to application variables; in the future, we might use a different approach to configuring things, but that'll remain being the same API.

Let's test things.

As written in `riak_ensemble_peer.erl`:

``` erlang
reset_follower_timer(State) ->
    set_timer(?FOLLOWER_TIMEOUT, follower_timeout, State).
```

What appears in `riak_ensemble_peer.P` after `erlc -I include -P src/riak_ensemble_peer.erl`:

``` erlang
reset_follower_timer(State) ->
    set_timer(riak_ensemble_config:follower_timeout(), follower_timeout, State).
```

Looks good.

Comment (reiddraper) [inline src/riak_ensemble_peer.erl:67]: Ah right, I was thinking the function was evaluated at compile time, but you're definitely right. Carry on.

**Comment by seancribbs** [comment]
:+1: r=andrewjstone

**Comment by andrewjstone** [comment]
@borshop merge

**Pull request merged**